dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageCheckOption.java

```
@@ -0,0 +1,103 @@
+    abstract Builder setJarFileList(List<Path> value);
+    abstract Builder setReportOnlyReachable(boolean value);
+    abstract StaticLinkageCheckOption build();
+  }
+
+  static StaticLinkageCheckOption parseArgument(String[] arguments) {
+    Options options = new Options();
+    options.addOption(
+        "b", "bom", true, "BOM to generate a classpath");
+    options.addOption(
+        "c", "coordinate", true, "Maven coordinates (separated by ',') to generate a classpath");
+    options.addOption("j", "jars", true, "Jar files (separated by ',') to generate a classpath");
+    options.addOption(
+        "r",
+        "report-only-reachable",
+        false,
+        "To report only linkage errors reachable from entry point");
+
+    CommandLineParser parser = new DefaultParser();
+    List<Path> jarFilePaths = new ArrayList<>();
+
+    ImmutableList.Builder<String> mavenCoordinates = ImmutableList.builder();
+    try {
+      CommandLine cmd = parser.parse(options, arguments);
+      if (cmd.hasOption("c")) {
+        String mavenCoordinatesOption = cmd.getOptionValue("c");
+        mavenCoordinates.addAll(Arrays.asList(mavenCoordinatesOption.split(",")));
+      }
+      if (cmd.hasOption("j")) {
+        String jarFiles = cmd.getOptionValue("j");
+        List<Path> jarFilesInArguments =
+            Arrays.stream(jarFiles.split(","))
+                .map(name -> (Paths.get(name)).toAbsolutePath())
+                .collect(Collectors.toList());
+        jarFilePaths.addAll(jarFilesInArguments);
+      }
+
+      String mavenBomCoordinate = cmd.getOptionValue("b");
+
+      boolean reportOnlyReachable = cmd.hasOption("r");
+
+      return builder()
+          .setBomCoordinate(mavenBomCoordinate)
+          .setMavenCoordinates(mavenCoordinates.build())
+          .setJarFileList(jarFilePaths)
+          .setReportOnlyReachable(reportOnlyReachable)
+          .build();
+    } catch (ParseException ex) {
```

Does it make sense to just let the ParseException be thrown instead?

---

I added helpFormetter.printHelp after catching ParseException.
Now ParseException is re-thrown.